src/main/java/PanoViewer/FileChooser.java

```diff
@@ -0,0 +1,28 @@
+package PanoViewer;
+
+import javax.imageio.ImageIO;
+import javax.swing.*;
+import javax.swing.filechooser.FileFilter;
+import javax.swing.filechooser.FileNameExtensionFilter;
+
+public class FileChooser {
+
+    private JFileChooser fileChooser;
+    private static FileChooser instance = new FileChooser();
+
+    public static FileChooser getInstance() {
+        return instance;
+    }
+
+    private FileChooser() {
+        String dir = System.getProperty("user.dir");
+        fileChooser = new JFileChooser(dir);
+        FileFilter fileFilter = new FileNameExtensionFilter("Images Files", ImageIO.getReaderFileSuffixes());
+        fileChooser.addChoosableFileFilter(fileFilter);
+        fileChooser.setFileFilter(fileFilter);
+    }
+
+    public JFileChooser chooser() {
+        return fileChooser;
```

Comment:
You should reset the filefilters each time you return the instance, so that the default filter would not change if the user changes them when they open the dialog(say they change it to open all files). This is also why you should cache the fileFilter.